### ISSUE TITLE
Support for Raspberry PI 240p over composite output

### ISSUE THREAD

**Issue by starquake** [libretro/picodrive#50]
Opening a new issue because I couldn't find an issue that describes mine. Or maybe I don't understand how it should work.

This is in issue about what (I think) should work and is lacking in a lot of cores. (I just picked Picodrive to start with.) I could have posted it as an issue in in RetroArch. Please tell me if I should have.

Hopefully I don't come across too harsh. I really love all the work that is put into libretro and all the cores. I want to make it great for people using 240p composite with a Raspberry PI too. Someone has to report the issues one will face when using the new 240p output.

Now that the Raspberry PI supports 240p output in the latest firmware I would think the following features would come in handy.

My CRT screen is calibrated to an AV Famicom and a Amiga and some other composite stuff. I wouldn't want to calibrate for just the Raspberry PI.

The Raspberry PI outputs at 720x480. When set to progressive (240p) it skips every other line (don't know if it's even or odd, but that doesn't really matter). The 720 width has some overscan. The viewable range on my television is about 660. What I would want is to get a picture from picodrive that is 660x480.

To get 480 I need 240p multiplied by 2. This means if the output is 224, (or even 192 in the case of SMS?) I need black parts on the top and bottom. I think this normally is accomplished by disabling overscan cropping. I can not find this in the pico drive core in the latest lakka (Lakka-RPi2.arm-8.0-devel-20170305031646-r25834-g76ebcbd.img)

For the width I would want it to scale to the available width. Even it means scaling is non-integer.

Much of this can be achieved by using a custom viewport. I can set it to 660x448 and it looks great. RGUI looks really ugly though because it scales to 240p. Also if I switch to other cores that do output 240p I have a problem with scaling.

What I would ultimately want is for (1) all the cores to output a width of 660, even if that means the scaling isn't integer. (With my CRT television non-integer scaling isn't that visible, I might add a filter that does horizontal filtering only to get rid of jagged edges)
(2) all the cores to output to a height of 480

This way I can put lakka in a mode of 660x480 and cores with 192x240, 256x240 and 320x240 all are displayed in a good way.

I hope you can help me or point me into the right direction. If I can help you please let me know!

tl;dr

It would be great if all the cores could output to something like 660x480 for use with 240p over composite, let's start with picodrive.

**Comment by jayare5** [comment]
I second this. I didn't know there would be this problem with multiple cores, I intend to use Lakka, but I haven't set up the composite option yet to try or make any suggestions :/.

**Comment by starquake** [comment]
After contemplating for a while and reading some sourcecode of different cores and libretro, I don't think fixing this issue should start at a core. Going to look some more into the code and see if I can work out a way to approach this.

**Comment by jayare5** [comment]
Oh man, I found a very similar issue to this.
When I play Game Gear, it uses the same core used to play Genesis.
If I set the resolution on my Lakka so that the Game Gear screen matches each line within the 240p mode (that is like 144 lines height?) then it will be perfect for Game Gear, but then loading a Genesis game will shrink it to that, which messes up all the lines.

I have to use one of the cores for Genesis games and the other for Game Gear xD............
This is gonna be messy.
Additionally, setting the Geme Gear and GameBoy resolutions to be 1:1 within 240p will mean that the menu will become tiny.
This also makes Sega Master System games mess up too. I would need a 3rd core to run all these 3 consoles successfully (unless there was an easy option to load settings automatically depending if it is SMS, Gen, or GG, but with so many options already, I doubt this would be pretty.)

The Lakka menu should have its own 240p setting regardless what console you're running.

**Comment by hizzlekizzle** [comment]
You can modify the sgb border shader to load other borders and match the resolution of whatever content you want:
https://github.com/hizzlekizzle/glsl-shaders/blob/master/borders/sgb/sgb.glslp
That is, you can change the in_res and out_res values to match your situation.

**Comment by jayare5** [comment]
Oh man, I'd love to be able to use this! I'm a big fan of having images around games like the Super GameBoy :D
I'm still a noob though, I'll need to research more.

**Comment by irixxxx** [comment]
@hizzlekizzle Can this be somehow moved to RA? I don't think it is core related.

**Comment by hizzlekizzle** [comment]
I don't know of any way to move it, but I think you can safely close it as a wontfix because 1.) yeah, it's not really a core issue and 2.) that's not something we're likely to ever be able to do, since the composite output is controlled by a closed-source firmware.
